backend/src/main/java/com/bakdata/conquery/models/auth/AuthorizationHelper.java

```diff
@@ -116,4 +124,44 @@ public static void addPermission(PermissionOwner<?> owner, ConqueryPermission pe
 	public static void removePermission(PermissionOwner<?> owner, Permission permission, MasterMetaStorage storage) throws JSONException {
 		Objects.requireNonNull(owner).removePermission(storage, permission);
 	}
+
+	/**
+	 * Returns a list of the effective permissions. These are the permissions of the
+	 * owner and the permission of the roles it inherits.
+	 * 
+	 * @param storage
+	 *
+	 * @return Owned and inherited permissions.
+	 */
+	public static Set<ConqueryPermission> getEffectiveUserPermissions(UserId userId, MasterMetaStorage storage) {
+		User user = storage.getUser(userId);
+		Set<ConqueryPermission> permissions = new HashSet<>(user.getPermissions());
+		for (Role role : user.getRoles()) {
+			permissions.addAll(role.getPermissions());
+		}
+
+		for (Group group : storage.getAllGroups()) {
+			if (group.containsMember(user)) {
+				// Get Permissions of the group
+				permissions.addAll(group.getPermissions());
+				// And all of all roles a group holds
+				group.getRoles().forEach(r -> permissions.addAll(r.getPermissions()));
+			}
+		}
+		return permissions;
+	}
+
+	/**
+	 * Returns the groups a user belongs to.
+	 */
+	public static Collection<Group> getGroupsOf(User user, MasterMetaStorage storage) {
+		Collection<Group> allGroups = storage.getAllGroups();
+		List<Group> userGroups = new ArrayList<>();
+		allGroups.forEach(g -> {
```

Comment:
würde hier normales for(:) machen stattdessen

Reply:
oki